### ISSUE TITLE
JavaScript library must check input arguments validity

### ISSUE THREAD

**Issue by sbordet** [sbordet/import_issues#35]
It is very easy to make the server fail when, from the JS client, calls to the public methods are made without the required parameters.

`cometd.publish(); // Missing channel and data`

results in:

```
java.lang.NullPointerException
	at org.cometd.server.AbstractBayeux.handle(AbstractBayeux.java:282)
```
and 

`cometd.publish({}); // Missing channel`

results in:

```
java.lang.Error
	at org.cometd.server.AbstractCometdServlet.getMessages(AbstractCometdServlet.java:322)
	...
	at java.lang.Thread.run(Thread.java:619)
Caused by: java.lang.ClassCastException: java.util.HashMap cannot be cast to java.lang.String
	at org.cometd.server.MessageImpl.put(MessageImpl.java:258)
```
And a moment later the connect fails with a JSON parsing error (this may be another bug in the server side):

```
java.lang.Error
	at org.cometd.server.AbstractCometdServlet.getMessages(AbstractCometdServlet.java:322)
	....
	at java.lang.Thread.run(Thread.java:619)
Caused by: java.lang.IllegalStateException: unknown char ','(44) in org.eclipse.jetty.util.ajax.JSON$ReaderSource@3085a863
	at org.eclipse.jetty.util.ajax.JSON.handleUnknown(JSON.java:807)
```
Checks should be performed in the client to avoid these problems, as people monitor their server and can worry seeing these errors.

**Comment by sbordet** [comment]
Originally reported by sbordet on 2009-10-23 04:58:49.0

**Comment by sbordet** [comment]
sbordet on 2009-11-16 11:00:58.0:
Fixed by adding the relevant checks on the JS client library.
The last example also showed a server-side bug that's been addressed in Jetty 7.0.1, which cometd-1.0 will depend on.
